src/io/flutter/logging/FlutterConsoleLogManager.java

```diff
@@ -126,95 +127,157 @@ public void handleFlutterErrorEvent(@NotNull Event event) {
     final String description = " " + diagnosticsNode.toString() + " ";
 
-    final int targetLength = errorSeparator.length();
-    final String prefix = StringUtil.repeat("◢◤", (targetLength - description.length()) / 4);
-    final String suffix = StringUtil.repeat("◢◤", ((targetLength - (description.length() + prefix.length())) / 2));
+    final boolean terseError = errorsSinceReload != 0;
 
-    console.print("\n" + prefix + description + suffix + "\n", TITLE_CONTENT_TYPE);
+    // TODO(devoncarew): Potentially change the error separator chars based on the error type (overflow, ...).
+    final String prefix = "════════";
+    final String suffix = "══";
 
-    DiagnosticLevel lastLevel = null;
+    console.print("\n" + prefix, TITLE_CONTENT_TYPE);
+    console.print(description, NORMAL_CONTENT_TYPE);
+    console.print(
+      StringUtil.repeat(errorSeparatorChar, Math.max(errorSeparatorLength - prefix.length() - description.length() - suffix.length(), 0)),
+      TITLE_CONTENT_TYPE);
+    console.print(suffix + "\n", TITLE_CONTENT_TYPE);
 
     // TODO(devoncarew): Create a hyperlink to a widget - ala 'widget://inspector-1347'.
 
-    for (DiagnosticsNode property : diagnosticsNode.getInlineProperties()) {
-      // Add blank line between hint and non-hint properties.
-      if (lastLevel != null && lastLevel != property.getLevel()) {
-        if (lastLevel == DiagnosticLevel.hint || property.getLevel() == DiagnosticLevel.hint) {
-          console.print("\n", NORMAL_CONTENT_TYPE);
+    if (terseError) {
+      for (DiagnosticsNode property : diagnosticsNode.getInlineProperties()) {
+        printTerseNodeProperty(console, "", property);
+      }
+    }
+    else {
+      DiagnosticLevel lastLevel = null;
+
+      for (DiagnosticsNode property : diagnosticsNode.getInlineProperties()) {
+        // Add blank line between hint and non-hint properties.
+        if (lastLevel != null && lastLevel != property.getLevel()) {
+          if (lastLevel == DiagnosticLevel.hint || property.getLevel() == DiagnosticLevel.hint) {
+            console.print("\n", NORMAL_CONTENT_TYPE);
+          }
         }
+
+        lastLevel = property.getLevel();
+
+        printDiagnosticsNodeProperty(console, "", property, null, false);
       }
+    }
 
-      lastLevel = property.getLevel();
+    console.print(StringUtil.repeat(errorSeparatorChar, errorSeparatorLength) + "\n\n", TITLE_CONTENT_TYPE);
+  }
 
-      printDiagnosticsNodeProperty(console, "", property, null, false);
+  private void printTerseNodeProperty(ConsoleView console, String indent, DiagnosticsNode property) {
+    boolean skip = true;
+
+    if (property.getLevel() == DiagnosticLevel.summary) {
+      skip = false;
+    }
+    else if (property.hasChildren()) {
+      try {
+        final CompletableFuture<ArrayList<DiagnosticsNode>> future = property.getChildren();
+        final ArrayList<DiagnosticsNode> children = future.get();
```

Comment:
you could all
future.getNow(null)
the children are already present for the error message payload case.
Calling .get() could lock. This will eliminate catching the threading exceptions.

Reply:
ah, great, will make this change